package/MDAnalysis/analysis/MSD.py

```diff
@@ -0,0 +1,468 @@
+
+MSD command line usage
+----------------------
+
+This calculates the MSD of 3 atom selections over the first 50000 frames
+of the trajectory:
+
+    >>> ./MSD.py -f mysim.xtc -s mysim.tpr --sel 'name NA,name NA and prop z
+            > 50 and prop z < 150, name CL' -b 0 -e 50000 --dt 5 -o mysim.p
+            --len 1500
+
+Output will be written to mysim.p in the format [msd,n_samples], where msd
+is a vector of length=3 containing vectors of MSD values with index
+corresponding to time interval, and n_samples is a similar array containing
+the number of samples associated with each index in msd.
+
+The option --len alters the number of calculations performed by the tool. If
+you are taking a linear regression of the MSD vs tau plot over the first 'n'
+frames, adjusting this parameter allows you to collect the data you need,
+allowing you to avoid calculating MSD values at tau >= n+1.
+
+The option --max_data will use data after the frame specified by -e if it is
+available. This allows you to avoid losing data if you need to split the
+calculation over several parallel jobs.
+
+See the main method of MSD.py for an example of python script implementation.
+
+Options:
+--------
+-f name of trajectory file
+-s name of tpr file
+--sel string containing selections separated by commas
+-b t0 (frames)
+-e tf (frames)
+--dt number of frames between restarts
+-o output file name (optional)
+--len maximum number of tau frames to calculate (optional, default=250)
+--out write output boolean (optional, default=True)
+--max_data include data from after tf if it exists
+  this allows for command-line level parallelization without
+  data loss. (optional, default=False)
+"""
+
+
+from __future__ import absolute_import, print_function, division
+
+import sys
+import os
+import getopt
+import warnings
+from collections import deque
+
+import numpy as np
+import pickle
+# from six import range
+
+import MDAnalysis
+from MDAnalysis.lib.distances import squared_distance_vector
+
+
+class MSD(object):
+    """Object for calculating mean square displacement (MSD)
+
+    This object can be used to accurately calculate the MSD of dynamic
+    (or static) selections (for example: name CL and z > 50, etc.).
+    The variables len_msd and dt_restart can be used to mitigate
+    computational and memory costs. This allows calculations to scale
+    well with simulation time and size--though you must have
+    enough RAM to load the trajectory during calculation.
+
+    The algorithm uses all available data in calculation, as the
+    calculation incorporates all atoms which have remained in the
+    dynamic selection over the interval, tau, from the restart.
+
+    Parameters
+    ----------
+    universe : object
+        Universe object containing trajectory data
+    select_list : list
+        List of selection strings
+    t0 : int
+        Time when analysis begins (in frames)
+    tf : int
```

Comment:
please use more meaningful variable names like `stop_frame`. The two letters `tf` can have a large number of meanings and it makes the code where you actually use the variables hard to understand.